libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1444,8 +1444,7 @@ void TypeChecker::checkExpressionAssignment(Type const& _type, Expression const&
 		auto const* tupleType = dynamic_cast<TupleType const*>(&_type);
 		auto const& types = tupleType ? tupleType->components() : vector<TypePointer> { _type.shared_from_this() };
 
-		solAssert(tupleExpression->components().size() == types.size(), "");
-		for (size_t i = 0; i < types.size(); i++)
```

Comment:
Looks good! Please add an assertion about `m_errorReporter.hasErrors()` for safety.

Reply:
Many thanks @chriseth. That's a good one.

Reply:
@chriseth: done.